lib/dealership.rb

```diff
@@ -0,0 +1,42 @@
+class Dealership
+  attr_reader :name, :address, :inventory
+
+  def initialize(name, address)
+    @name = name
+    @address = address
+    @inventory = []
+  end
+
+  def inventory_count
+    @inventory.size
+  end
+
+  def add_car(car)
+    @inventory << car
+  end
+
+  def has_inventory?
+    @inventory.size > 0
+  end
+
+  def cars_by_make(make)
+    @inventory.select do |car|
+      car.make == make
+    end
+  end
+
+  def total_value
+    @inventory.map do |car|
+      car.total_cost
+    end.sum
+  end
+
+  def average_price_of_car
+    (total_value / inventory.size).to_s.insert(-4, ",")
```

Comment:
Nice job solving this. I think this works for this challenge. I also solved it this way but I got to thinking that it might not work if we had a six figure number.